users/models.py

```diff
@@ -94,3 +141,19 @@ def last_activity_ago(self: "User") -> str:
             result = f"{delta.days} days ago"
 
         return result
+    
+    @property
+    def nickname(self):
+        return self.profile.nickname if self.profile else "not-yet-named"
+
+    @nickname.setter
+    def nickname(self, value):
+        if self.profile:
```

Comment:
Since User model is the core model, i would by my intuition put OneToOne field not in User but in Profile and UserSettings. Then I'm wondering if it is not better to modify _create_user method and just add on last 2 lines these: Profile.objects.create(user=user, nickname=nickname)
UserSettings.objects.create(user=user, ... )
then you can remove nickname.setter from User model and you don't need anymore signals. Also since nickname is obligatory during registration im not sure if you need to get it anymore from email